src/Message/CompletePurchaseResponse.php

```diff
@@ -28,31 +28,33 @@ public function __construct(RequestInterface $request, $data)
         $this->data    = $data;
 
         if (strtolower($this->getSignatureValue()) !== $this->generateSignature()) {
-            throw new InvalidResponseException('Invalid hash');
+            if (strtolower($this->getSignatureValue(true) !== $this->generateSignature())) {
+                throw new InvalidResponseException('Invalid hash');
+            }
         }
     }
 
-    public function generateSignature()
+    public function generateSignature($currency = false): string
```

Comment:
```suggestion
    public function generateSignature(bool $includeCurrency = false): string
```

1. Типизация свойств
2. Булевые переменные нужно называть соответствующе. Не "валюта", а "учитывать валюту"